### PR TITLE
Move front-end to NextJS

### PR DESCRIPTION
**Pull request by kakarot-dev**
## Description
Move frontend to NextJS and store settings in SQL

### Todo
- [x] Move all frontend functionality to NextJS
- [x] Store settings in SQL and retrieve through Next with SSR
- [x] Finish tokens page

Note: This PR will not make a configuration tool for changing app settings, this will come in the future.

### CODE REVIEW

**Comment by crazyaccess** [comment]
Why not use vote react sir ?

**Comment by kakarot-dev** [comment]
> Why not use vote react sir ?

What is vote react? never heard of it
Next was riku's opinion, im just helping him out, Altho it feels like he's doing most of the work

File: client/.env [hunk 1]
```diff
@@ -1,4 +1,2 @@
-NEXT_PUBLIC_API_URL=/api
-NEXT_PUBLIC_APP_NAME=Backpack
-NEXT_PUBLIC_APP_DESCRIPTION=A file host for all your needs
+NEXT_PUBLIC_API_URL=http://localhost:3000/api
```
Comment (kakarot-dev) [inline client/.env:1]: Isnt this gonna be port 3001?

**Comment by kakarot-dev** [comment]
oh conflicts are resolved

File: Dockerfile [hunk 2]
```diff
@@ -16,4 +16,4 @@ RUN npm install && npm run build
 WORKDIR /usr/src
 COPY --from=builder /usr/src/backpack/target/release/backpack .
 
-ENTRYPOINT ["/bin/sh", "-c" , "node /usr/src/scripts/check.js && /usr/src/backpack /usr/src/client/build"]
\ No newline at end of file
+ENTRYPOINT ["/bin/sh", "-c" , "node /usr/src/scripts/check.js && /usr/src/backpack --client /usr/src/client/build"]
```
Comment (kakarot-dev) [inline Dockerfile:19]: We need to add a port due next's initialprops stuff
Idk if next export will work for this project

**Review by kakarot-dev — COMMENTED** [review]
Looks nice!

**Comment by kakarot-dev** [comment]
I think all most all the stuff needed to do are now done

Possible changes now might be

- Reversion of addition of deleteUrl
- Upgradation of version numbers to 1.0.0
- Removal of `black` and `white` from enums list of themes
- (.. adding to anything else riku wanted to do)

**Comment by JSH32** [comment]
For now, reverted delete URL. As for the docker thing I'm not going to worry about it too much so I'll update it later on when things are more solid

**Pull request merged**